reporter/cgroupv2.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package reporter // import "go.opentelemetry.io/ebpf-profiler/reporter"
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	lru "github.com/elastic/go-freelru"
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+)
+
+// lookupCgroupv2 returns the cgroupv2 ID for pid.
+func lookupCgroupv2(cgrouplru *lru.SyncedLRU[libpf.PID, string], pid libpf.PID) (string, error) {
```

Comment:
Shouldn't this be made public if it is meant to be "reused in other reporters"?

Reply:
It can. But then, as a public/stable method, I'm not sure it should be in the reporter package.
It can also be private for now, and be made public later on.

Reply:
Can you explain why this shouldn't go to `libpf/` when it's meant to be part of a public API?